sktime/utils/sklearn/_adapt_df.py

```diff
@@ -0,0 +1,31 @@
+"""Utility functions for adapting to sklearn."""
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+
+import numpy as np
+
+
+def prep_skl_df(df, copy_df=False):
+    """Make df compatible with sklearn input expectations.
+
+    Changes:
+    turns column index into a list of strings
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        list of indices to sample from
+    copy_df : bool, default=False
+        whether to mutate df or return a copy
+        if False, index of df is mutated
+        if True, original df is not mutated. If index is not a list of strings,
+        a copy is made and the copy is mutated. Otherwise, the original df is returned.
+    """
+    cols = df.columns
+    str_cols = cols.astype(str)
+
+    if not np.all(str_cols == cols):
```

Comment:
Minor comment: is this check required? Like if `df.columns` is modified always (in-place or not based on `copy_df`), will there be any issues?

Reply:
I worry about cases where the index type is different, but elements are the same.